src/java/org/apache/cassandra/db/Directories.java

```diff
@@ -111,7 +111,7 @@ public class Directories
     private static final Logger logger = LoggerFactory.getLogger(Directories.class);
 
     private static final double MAX_COMPACTION_DISK_USAGE = System.getProperty("palantir_cassandra.max_compaction_disk_usage") == null
-                                                            ? 0.95
+                                                            ? 0.90
```

Comment:
what's the context for this change?

Reply:
This setting dictates how much of the available space we can use for a single compaction. 0.95 means that, at most, disk will reach 95% usage as the compaction takes place. The setting doesn't however account for backups via snapshots. Its effect is at the start of the compaction for scoping how many of the existing sstables for the specified ks.cf being compacted can be part of the compaction. If there are too many sstables to start the compaction and stay below 95% multiple passes will be done with some sstables being removed out of the initial pass.

The problem with 95% is, its too high. It will lead to disk filling up more often than not for clusters that are above 80% that have any sort of single big column family. This is mainly due to snapshotting taking place as the compaction is in flight.